File: Aula_05/c++/02.cpp
#include <iostream>

class Viagem{
    private: 
    double distancia = 0, tempo = 0;

    public:
    void setDistancia (double v) {
        if (v > 0){
            distancia = v;
        } else {
            throw std::invalid_argument ("a distancia nao pode ser negativa");
        }
    }

    void setTempo (double v){
        if (v > 0){
            tempo = v;
        } else {
            throw std::invalid_argument ("o tempo nao pode ser negativo");
        }
    }

    double getDistancia () {
        return distancia;
    }

    double getTempo () {
        return tempo;
    }

    double velocidadeMedia () {
        return distancia / tempo;
    }
};

class UI{
    public:
    static void main(){
        Viagem x;

        double v;
        std::cout << "Informe o valor da distancia da viagem: ";
        std::cin >> v;
        x.setDistancia(v);

        std::cout << "Informe o valor do tempo da viagem: ";
        std::cin >> v;
        x.setTempo(v);

        std::cout << "Distancia = " << x.getDistancia() << " Tempo = " << x.getTempo() << std::endl;
        std::cout << "Velocidade media = " << x.velocidadeMedia() << std::endl;
    }
};

int main(){
    UI::main();
    return 0;
}

File: Aula_05/python/02.py
class Viagem:
    def __init__ (self):
        self.__distancia = 0
        self.__tempo = 0

    def set_distancia (self, v):
        if v > 0:
            self.__distancia = v
        else:
            raise ValueError ("a distancia nao pode ser negativa")
        
    def set_tempo (self, v):
        if v > 0:
            self.__tempo = v
        else:
            raise ValueError ("o tempo nao pode ser negativo")
        
    def get_distancia (self):
        return self.__distancia
    
    def get_tempo (self):
        return self.__tempo
    
    def velocidade_media (self):
        return self.__distancia / self.__tempo
    

class UI:
    @staticmethod
    def main():
        x = Viagem()

        x.set_distancia(float(input("Informe o valor da distancia da viagem: ")))
        x.set_tempo(float(input("Informe o valor do tempo da viagem: ")))

        print(f"Distancia = {x.get_distancia()} Tempo = {x.get_tempo()}")
        print(f"Velocidade media = {x.velocidade_media()}")

UI.main()